lua-aws/core.lua

```diff
@@ -86,11 +87,16 @@ local AWS = class.AWS {
 				method = 'GET'
 			})
 			local obj = self._json_engine.decode(resp.body)
-			config.accessKeyId = obj.AccessKeyId
-			config.secretAccessKey = obj.SecretAccessKey
-			config.sessionToken = obj.Token
+			config.credentials = Credentials.new({
+				accessKeyId = obj.AccessKeyId,
+				secretAccessKey = obj.SecretAccessKey,
+				sessionToken = obj.Token,
+			})
+		end
+		if config.accessKeyId and config.secretAccessKey then
```

Comment:
can you make this `elseif`? because at current master, if all of `role` and `accessKey`, `secretKey` are set (maybe mistakenly though), credentials from role will be used.